mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -3620,15 +3624,19 @@ struct ThreadwiseGemmRewritePattern
     SmallVector<Value, 2> memIndicesKN;
     extractForInductionVars({loopK, loopN}, &memIndicesKN);
     auto gemmBKN = b.create<AffineLoadOp>(loc, gemmB, memIndicesKN);
-    auto mul = b.create<MulFOp>(loc, b.getF32Type(), gemmAKM, gemmBKN);
 
+    Value mul = b.create<MulFOp>(loc, b.getF32Type(), gemmAKM, gemmBKN);
+    if (dataType == b.getIntegerType(16))
+      mul = b.create<MulIOp>(loc, b.getIntegerType(16), gemmAKM, gemmBKN);
```

Comment:
this logic is dangerous. you are adding dangling values into the compiled code. also, it doesn't handle f16 case.